provider/client.cpp

```diff
@@ -5,23 +5,92 @@
 
 #include "debug_tags.hpp"
 #include "mi_context.hpp"
+#include "mi_module.hpp"
 #include "mi_schema.hpp"
 #include "shared_protocol.hpp"
+#include "socket_wrapper.hpp"
+
+
+#include <cstdlib>
+#include <errno.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
```

Comment:
```
client.cpp:66:34: error: ‘close’ was not declared in this scope
                         close (fd);
```
You're missing ```#include <unistd.h>```, close is declared there. Otherwise the code won't compile.